devito/ir/iet/nodes.py

```diff
@@ -337,6 +337,19 @@ def writes(self):
         return self._writes
 
 
+class FuncPtrCall(Call):
+
+    """
+    Function Pointer Call.
+    """
+    def __init__(self, name, return_type, parameter_type, **kwargs):
+
+        super().__init__(name=name)
+
+        self.return_type = return_type
+        self.parameter_type = parameter_type
```

Comment:
parameterS ?
or simply "signature" or "param_types"? 

use `as_tuple` for flexibility